### PR TITLE
POST /setup handler. r=francisco

### PR DESCRIPTION
**Pull request by ferjm**

### CODE REVIEW

**Comment by ferjm** [comment]
r? @arcturus 

The tests can be improved significantly, but I'd like to do that in a follow up once I get to a point where the basic login feature is done.

File: src/users_router.rs [hunk 1]
```diff
@@ -86,10 +89,55 @@ impl UsersRouter {
...
+    fn setup(req: &mut Request) -> IronResult<Response> {
+        #[derive(RustcDecodable, Debug)]
+        struct SetupBody {
+            username: String,
+            email: String,
+            password: String
+        }
+
+        let mut payload = String::new();
+        req.body.read_to_string(&mut payload).unwrap();
```
Comment (arcturus) [inline src/users_router.rs:101]: Can this cause a panic if the payload is malformed, or binary or is not present?

File: src/users_router.rs [hunk 2]
```diff
@@ -86,10 +89,55 @@ impl UsersRouter {
...
+        let body: SetupBody = match json::decode(&payload) {
+            Ok(body) => body,
+            Err(error) => {
+                println!("{:?}", error);
```
Comment (arcturus) [inline src/users_router.rs:105]: Oh, there we are :)

**Comment by arcturus** [comment]
LGTM!

**Comment by ferjm** [comment]
Closes #9

**Pull request merged**